config/src/keys.rs

```diff
@@ -1,96 +1,42 @@
 // Copyright (c) The Libra Core Contributors
 // SPDX-License-Identifier: Apache-2.0
 
-use libra_crypto::{PrivateKey, Uniform, ValidKeyStringExt};
-use mirai_annotations::verify_unreachable;
-use serde::{de::DeserializeOwned, Deserialize, Deserializer, Serialize, Serializer};
-
-#[derive(Debug, PartialEq)]
-#[cfg_attr(any(test, feature = "fuzzing"), derive(Clone))]
-pub enum PrivateKeyContainer<T> {
-    Present(T),
-    Removed,
-    Absent,
-}
-
-impl<T> PrivateKeyContainer<T>
-where
-    T: PrivateKey,
-{
-    pub fn take(&mut self) -> Option<T> {
-        match self {
-            PrivateKeyContainer::Present(_) => {
-                let key = std::mem::replace(self, PrivateKeyContainer::Removed);
-                match key {
-                    PrivateKeyContainer::Present(priv_key) => Some(priv_key),
-                    _ => verify_unreachable!(
-                        "mem::replace returned value for PrivateKeyContainer different from the one observed right before the call"
-                    ),
-                }
-            }
-            _ => None,
-        }
-    }
-
-    pub fn public_key(&self) -> Option<T::PublicKeyMaterial> {
-        match self {
-            PrivateKeyContainer::Present(private_key) => Some(private_key.public_key()),
-            _ => None,
-        }
-    }
-}
+//! This file implements a KeyPair data structure.
+//!
+//! The point of a KeyPair is to deserialize a private key into a structure
+//! that will only allow the private key to be moved out once
+//! (hence providing good key hygiene)
```

Comment:
frankly unless we want to eliminate the clone constraint in the crypto library, we have no choice...